### PR TITLE
Add Braintree.Subscription module

### PR DESCRIPTION
**Pull request by radar**
It looks like we started on the effort of building a client at the same time! Mine is at https://github.com/radar/braintree_elixir. I'm a noob when it comes to Elixir and I like your code better. So I thought I should contribute to yours and probably "retire" mine.

### CODE REVIEW

**Comment by sorentwo** [comment]
@radar: I'm glad you found the project, and I appreciate the PR. The integration tests rely on private env vars, so I'm not entirely sure how to get them passing on Travis.

File: lib/subscription.ex [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+      {:ok, subscription} = Braintree.Subscription.create(%{
+      })
+
+      customer.company # Braintree
+  """
+  @spec create(Map.t) :: {:ok, t} | {:error, Error.t}
+  def create(params \\ %{}) do
+    case HTTP.post("subscriptions", %{subscription: params}) do
+      {:ok, %{"subscription" => subscription}} ->
+        subscription |> atomize |> Map.keys |> IO.inspect
```
Comment (sorentwo) [inline lib/subscription.ex]: I think you'll want to remove the inspect here.

File: test/integration/subscription_test.exs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  @moduletag :integration
+
+  alias Braintree.Customer
+  alias Braintree.Subscription
+
+  test "create/1 with a plan_id" do
+    assert {:ok, customer} = Customer.create(%{payment_method_nonce: "fake-valid-nonce"})
+    card = customer.credit_cards |> List.first
+    assert {:ok, subscription } = Subscription.create(%{payment_method_token: card.token, plan_id: "starter"})
+    IO.inspect subscription
```
Comment (sorentwo) [inline test/integration/subscription_test.exs]: And here, along with the comment.

**Comment by radar** [comment]
I've removed the debugging stuff now.

WRT to the integration tests, you could setup some encrypted variables on Travis: https://docs.travis-ci.com/user/environment-variables/#Encrypted-Variables.

For this test to pass, you'll need a plan with ID of "starter" in your sandbox account too.

**Comment by sorentwo** [comment]
AH, makes sense now. I have set up encrypted variables, it would probably be that the sandbox account doesn't have any plans.

**Comment by sorentwo** [comment]
I finally got the encrypted variables into `.travis.yml` and rebased this branch locally. Thanks!

**Pull request closed without merging**